src/main/assets/js/courts-table-search.ts

```diff
@@ -0,0 +1,220 @@
+        $(this.searchCourtsFilterId).val() as string,
+        $(`#main-content input[name=${this.toggleClosedCourtsDisplay}]`).prop('checked'),
+        $(this.courtsNameAscToggleId).val() as string,
+        $(this.courtsUpdatedAscToggleId).val() as string);
+
+    filterName == 'name' ?
+      CourtsTableSearch.switchTableClasses($(this.courtsNameAscToggleId),
+        $(this.tableCourtsNameId), $(this.tableCourtsUpdatedId)) :
+      CourtsTableSearch.switchTableClasses($(this.courtsUpdatedAscToggleId),
+        $(this.tableCourtsUpdatedId), $(this.tableCourtsNameId));
+  }
+
+  /**
+   *
+   * Retrieves the filtered results from the toggles and search, and sets the html
+   * of the tbody element of the table accordingly
+   *
+   * @param courts the list of public court information
+   * @param searchFilterValue the search value from the 'search courts or tribunals' input field
+   * @param includeClosedCourts whether or not to include closed courts in the result
+   * @param orderNameAscendingFilter whether to sort by name asc or desc
+   * @param orderUpdatedAscendingFilter whether to sort by last updated date asc or desc
+   * @private
+   */
+  private static setUpTableData(courts: string, searchFilterValue: string, includeClosedCourts: boolean,
+    orderNameAscendingFilter: string, orderUpdatedAscendingFilter: string): void {
+    const filteredCourts = CourtsTableSearch.filterCourts(courts, searchFilterValue, includeClosedCourts,
+      orderNameAscendingFilter, orderUpdatedAscendingFilter);
+    $(this.courtsResultsSection).html(CourtsTableSearch.getCourtsTableBody(filteredCourts));
+    searchFilterValue.length ? $(this.numberOfCourts).show().text(
+      'Showing ' + filteredCourts.length + ' results')
+      : $(this.numberOfCourts).hide();
+  }
+
+  /**
+   *
+   * Takes a list of courts, and filters based on a set list of criteria:
+   * 1. Name Asc/Desc
+   * 2. Updated date Asc/Desc
+   * 3. Resets whichever one is not enabled
+   * 4. Based on the above, also filters the result list based on the search string from the 'search' input field
+   *
+   * @param courts the list of public courts
+   * @param searchFilterValue the string value of the search input field
+   * @param includeClosedCourts whether or not to include closed courts in the result
+   * @param orderNameAscendingFilter whether to sort by name asc or desc
+   * @param orderUpdatedAscendingFilter whether to sort by last updated date asc or desc
+   * @private
+   */
+  private static filterCourts(courts: string, searchFilterValue: string, includeClosedCourts: boolean,
+    orderNameAscendingFilter: string, orderUpdatedAscendingFilter: string): any {
+
+    return JSON.parse(courts)
+      .filter((court: { displayed: boolean }) => {
+        if (includeClosedCourts) {
+          return court.displayed == true || court.displayed == false;
+        }
+        else return court.displayed == true;
+      }).filter((court: { name: string }) => {
```

Comment:
Does filtering twice mean traversing the collection twice? In both filters, is there an existing court type that should be used?